content/blog/2024-08-02-unikraft-gsoc-lxboot_x86.mdx

```diff
@@ -0,0 +1,106 @@
+---
+title: "GSoC'24: Linux x86 Boot Protocol Support"
+description: |
+  The Linux boot protocol plays an important role in the initialization of the Linux operating system, emphasizing the importance of system optimization and scalability.
+publishedDate: 2024-08-02
+image: /images/unikraft-gsoc24.png
+tags:
+- gsoc
+- gsoc24
+- booting
+authors:
+- Mihnea Firoiu
+---
+
+## Setup in Unikraft
+
+First of all, I added the following lines to the `lxboot.S` file from Unikraft:
+
+```diff
+diff --git a/plat/kvm/x86/lxboot.S b/plat/kvm/x86/lxboot.S
+index 0ebe18a5..8140804f 100644
+--- a/plat/kvm/x86/lxboot.S
++++ b/plat/kvm/x86/lxboot.S
+@@ -21,6 +21,13 @@ lcpu_boot_startup_args:
+        .quad   0
+        .quad   0
+ 
++.code32
```

Comment:
The font here is pretty bad, the green color is almost non visible for me.
Is it the same for you? If so, maybe we can find a way to change it, maybe replace `diff` with something else. If this is only on my side then it's file.

Reply:
For me the green color looks good, but I could change it to something else if you say so.

Reply:
It's fine, leave it.